main.js

```diff
@@ -92,41 +92,67 @@ function removeError(){
-    div_form_userName.setAttribute("id", "form_username");
-    div_form_userName.innerHTML = '<label for="username">Username:</label>' +
-    '<input type="text" name="username" id="username" placeholder="username">';
-
-    const div_form_password = form.appendChild(document.createElement("div"));
-    div_form_password.setAttribute("class", "form_input");
-    div_form_password.setAttribute("id", "form_password");
-    div_form_password.innerHTML = '<label for="password">Password:</label>' +
-    '<input type="password" name="password" id="password" placeholder="password">';
-
-    const div_button = form.appendChild(document.createElement("div"));
-    div_button.setAttribute("class", "button");
-    div_button.innerHTML = '<button id="login_btn">login</button>';
-    
-    document.body.append(div_form_container);
-    return div_form_container;
-}
-
+    document.body.append(createFormContainer());
 
+    const parentElement = document.getElementById("login");
+    parentElement.appendChild(createTitle("Login form"));
+    parentElement.appendChild(createInputUsername());
+    parentElement.appendChild(createInputPassword());
+    parentElement.appendChild(createButton("login_btn")); 
+}
 
+function createFormContainer(){
+    const div_form_container = createBaseElement({className: "form_container", id: ""});
+    div_form_container.appendChild(createBaseElement({element: "form", className: "form", id: "login"}));
+    return div_form_container;
+}
 
+function createInputUsername(){
+    const username = createBaseElement({className: "form_input", id: "form_username"});
+    username.appendChild(createLabel("username", "Username"));
+    username.appendChild(createInput("username"));
+    return username;
+}
 
+function createInputPassword(){
+    const password = createBaseElement({className: "form_input", id: "form_password"});
+    password.appendChild(createLabel("password", "Password"));
+    password.appendChild(createInput("password"));
+    return password;
+}
 
+function createBaseElement({element = "div", className, id}){
+    const new_element = document.createElement(element);
+    if(className) new_element.setAttribute("class", className);
+    if(id) new_element.setAttribute("id", id);
+    return new_element;
+}
 
+function createButton(id){
+    const div_button = createBaseElement({className: "button", id: ""});
+    const button = createBaseElement({element: "button", className: "", id: id});
+    button.textContent = "login";
+    div_button.appendChild(button);
+    return div_button;
+}
 
+function createTitle(text){
+    const title = document.createElement("h1");
+    title.textContent = text;
+    return title;
+}
 
+function createLabel(htmlFor, text){
+    const label = document.createElement("label");
+    label.htmlFor = htmlFor;
+    label.textContent = text;
+    return label;
+} 
+
+function createInput(id){
+    const username = createBaseElement({element: "input", className: "", id: id});
```

Comment:
`className: ""` - if I check this code for a boolean value, then I will get a `false`. So you don't have to pass this value in your config object. 

Because of `if(className)` in the `createBaseElement` function

Reply:
You can leave it as `createBaseElement({element: "input", id: id})`

Reply:
Same thing with `id`

Reply:
Done